builder.ts

```diff
@@ -289,24 +304,27 @@ export class PackDefinitionBuilder implements BasicPackDefinition {
    * ```
    */
   setSystemAuthentication(systemAuthentication: SystemAuthenticationDef): this {
-    const {
-      getConnectionName: getConnectionNameDef,
-      getConnectionUserId: getConnectionUserIdDef,
-      postSetup: postSetupDef,
-      ...rest
-    } = systemAuthentication;
-    const getConnectionName = wrapMetadataFunction(getConnectionNameDef);
-    const getConnectionUserId = wrapMetadataFunction(getConnectionUserIdDef);
-    const postSetup = postSetupDef?.map(step => {
-      return {...step, getOptions: wrapMetadataFunction(setEndpointDefHelper(step).getOptions)};
-    });
-    this.systemConnectionAuthentication = {
-      ...rest,
-      getConnectionName,
-      getConnectionUserId,
-      postSetup,
-    } as SystemAuthentication;
+    // TODO(patrick): Remove this cast
+    this.systemConnectionAuthentication = this._wrapAuthenticationFunctions(
+      systemAuthentication,
+    ) as SystemAuthentication;
+    return this;
+  }
 
+  /**
+   * TODO(patrick): Unhide this
+   * @hidden
+   */
+  setAdminAuthentications(adminAuthentications: AdminAuthentication[]): this {
```

Comment:
In the builder pattern, we've used `set()` to mean "set or overwrite a singular value" and `add()` to mean "append to an array value", so I think this should be `addAdminAuthentication()` and take a single value and not an array.

Reply:
Done